azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VaultCertificate.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.json.SerializedNames;
+
+/**
+ * Certificate stored in a Key Vault
+ */
+@AutoValue
+public abstract class VaultCertificate {
+
+    /**
+     * The URL of the certificate
+     */
+    public abstract String certificateUrl();
+
+    /**
+     * Certificate's store name
+     */
+    public abstract String certificateStore();
```

Comment:
Needs `@Nullable` - both here, and on the arg to `create` - see https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-create-or-update#bk_vaultcert, which says that it is not required.

I believe that if you don't include `@Nullable` then `AutoValue`'s auto-generated code will throw a `NullPointerException` if the value is null.

(Actually, probably don't need it on the arg to `create()` - we don't do that elsewhere).